packages/meditrak-server/src/routes/importSurveyResponses/SurveyResponseUpdatePersistor.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+const RECORDS_PER_BULK_BATCH = 5000; // number of records (survey responses + answers) processed per bulk insert/delete
+
+export const CREATE = 'create';
+export const UPDATE = 'update';
+export const DELETE = 'delete';
+
+function batchCreates(creates) {
+  const batches = [[]];
+  let currentBatchIndex = 0;
+  let currentBatchRecordCount = 0;
+  for (const create of creates) {
+    const answerCount = create.answers.upserts.length;
+    const recordCount = answerCount + 1; // +1 for the response itself
+    if (currentBatchRecordCount + recordCount > RECORDS_PER_BULK_BATCH) {
+      batches.push([]);
+      currentBatchIndex++;
+      currentBatchRecordCount = 0;
+    }
+    batches[currentBatchIndex].push(create);
+    currentBatchRecordCount += recordCount;
+  }
+  return batches;
+}
+
+function batchDeletes(deletes) {
+  const batches = [];
+  for (let i = 0; i < deletes.length; i += RECORDS_PER_BULK_BATCH) {
+    batches.push(deletes.slice(i, i + RECORDS_PER_BULK_BATCH));
+  }
+  return batches;
```

Comment:
Yep I thought about it, but it wasn't being imported anywhere, and we needed custom logic for the create batching anyway, so decided just to write the (very small) function rather than adding the import. I do like `chunk` though

Reply:
Fair point!